warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java

```diff
@@ -69,9 +69,8 @@ public Object visit(ASTFunctionNode node, Object data) {
                 }
             }
             
-            // only re-parent if the same number of regex nodes were built
             if (identifiers.size() == children.size()) {
-                switch (identifiers.size()) {
+                switch (children.size()) {
```

Comment:
So 6 months ago you thought we should switch on `identifiers.size()`, but now you're had a change of heart?  Does this need to be changed?  Also, I would keep the comment.

Reply:
Good catch, that's a failure to fully revert. I suppose it doesn't matter since the collections are the same size, but I'll keep it consistent with what was there before.